src/citrine/informatics/design_spaces/hierarchical_design_space.py

```diff
@@ -0,0 +1,215 @@
+
+    def __init__(
+            self,
+            *,
+            material_template: UUID,
+            process_template: UUID,
+            template_name: Optional[str] = None
+    ):
+        self.material_template: UUID = material_template
+        self.process_template: UUID = process_template
+        self.template_name: Optional[str] = template_name
+
+
+class MaterialNodeDefinition(Serializable["MaterialNodeDefinition"]):
+    """A single node in a material history design space.
+
+    Parameters
+    ----------
+    name: str
+        A unique name used to reference the materials produced by this node in the design space.
+        This name should be used to identify the material when used as an ingredient
+        in the formulation subspace of a different node.
+    scope: Optional[str]
+        An optional custom scope used to identify the materials produced by this node.
+    attributes: List[Dimension]
+        Set of dimensions included on materials produced by this node.
+    formulation_subspace: Optional[FormulationDesignSpace]
+        An optional formulation design space defining the ingredients, labels,
+        and constraints for formulations in materials produced by this node.
+    template_link: Optional[TemplateLink]
+        A set of identifiers linking the materials produced by this node to
+        material and process templates on the Citrine Platform.
+    display_name: Optional[str]
+        Optional name for identifying the node on the Citrine Platform.
+        Solely for display purposes (does not appear in generated hierarchical candidates)
+
+    """
+
+    name = properties.String("identifier.id")
+    scope = properties.Optional(properties.String, "identifier.scope")
+    attributes = properties.List(properties.Object(Dimension), "attributes")
+    formulation_subspace = properties.Optional(
+        properties.Object(FormulationDesignSpace), "formulation"
+    )
+    template_link = properties.Optional(properties.Object(TemplateLink), "template")
+    display_name = properties.Optional(properties.String, "display_name")
+
+    def __init__(
+            self,
+            *,
+            name: str,
+            scope: Optional[str] = None,
+            attributes: Optional[List[Dimension]] = None,
+            formulation_subspace: Optional[FormulationDesignSpace] = None,
+            template_link: Optional[TemplateLink] = None,
+            display_name: Optional[str] = None
+    ):
+        self.name = name
+        self.scope: Optional[str] = scope
+        self.attributes = attributes or list()
+        self.formulation_subspace: Optional[FormulationDesignSpace] = formulation_subspace
+        self.template_link: Optional[TemplateLink] = template_link
+        self.display_name: Optional[str] = display_name
+
+    def __repr__(self):
+        display_name = self.display_name or self.name
+        return "<MaterialNodeDefinition {!r}>".format(display_name)
```

Comment:
```suggestion
        return f"<MaterialNodeDefinition {display_name!r}>"
```
Not required, but I prefer f-strings from a legibility perspective.